### PR TITLE
fix(editor): Fix issue that double incoming connection often does not resolve expression (no-changelog)

### PR DESCRIPTION
**Pull request by alexgrozav**
Cherry-picked from AI Tool Creation branch

https://community.n8n.io/t/error-can-t-get-data-for-expression-on-set-node-variable-preview-when-connected-to-both-ends-of-if-node/26495/28

### CODE REVIEW

File: packages/editor-ui/src/mixins/workflowHelpers.ts [hunk 1]
```diff
@@ -317,12 +317,10 @@ function executeData(
...
 
-	if (parentNode.length) {
-		// Add the input data to be able to also resolve the short expression format
-		// which does not use the node name
-		const parentNodeName = parentNode[0];
-		const workflowsStore = useWorkflowsStore();
+	const workflowsStore = useWorkflowsStore();
 
+	// Find the parent node which has data
+	for (const parentNodeName of parentNodes) {
```
Comment (mutdmour) [inline packages/editor-ui/src/mixins/workflowHelpers.ts:323]: can you please add tests?

**Review by mutdmour — CHANGES_REQUESTED** [review]

File: packages/editor-ui/src/mixins/workflowHelpers.ts (hunk 1 above)
Comment (alexgrozav) [reply packages/editor-ui/src/mixins/workflowHelpers.ts:323]: Added tests. Thanks!

**Review by mutdmour — DISMISSED** [review]
LGTM.

File: packages/editor-ui/src/mixins/workflowHelpers.ts [hunk 2]
```diff
@@ -49,7 +49,7 @@ import { nodeHelpers } from '@/mixins/nodeHelpers';
 import { genericHelpers } from '@/mixins/genericHelpers';
 import { useToast, useMessage } from '@/composables';
 
-import { isEqual } from 'lodash-es';
+import { get, isEqual } from 'lodash-es';
```
[the remark this answers is not in the document]
Comment (alexgrozav) [reply packages/editor-ui/src/mixins/workflowHelpers.ts]: Removed. Thanks!

**Review by mutdmour — APPROVED** [review]

**Pull request merged**